pkgs/development/compilers/llvm/16/default.nix

```diff
@@ -253,7 +253,18 @@ in let
       nixSupport.cc-cflags =
         [ "-rtlib=compiler-rt"
           "-Wno-unused-command-line-argument"
+          # The compiler got stricter about this, so a number of downstream
+          # things seem to fail with it switched on.
+          "-Wno-implicit-function-declaration"
```

Comment:
This seems questionable to me. Those things should be fixed. Quite a few have been done already as part of #234710. I still have a few more PRs left to open, but everything in [this gist][1] should build with clang 16. Once #241692 is merged into staging, I expect there will be another round of fixes.

[1]: https://gist.github.com/reckenrode/bf3161814800572e8d38fee52ff3ae40